pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -254,6 +254,59 @@ func getAllDomainDisks(dom cli.VirDomain) ([]api.Disk, error) {
 	return devices.Disks, nil
 }
 
+func (l *LibvirtDomainManager) hotplugMemory(vmi *v1.VirtualMachineInstance, spec *api.DomainSpec, dom cli.VirDomain) error {
+
+	// check whether virt-launcher is ready for hotplug
+	if _, exists := vmi.Labels[v1.MemoryHotplugReadyLabel]; !exists {
+		return nil
+	}
+
+	if vmi.Spec.Domain.Memory == nil ||
+		vmi.Spec.Domain.Memory.Guest == nil ||
+		vmi.Spec.Domain.Memory.MaxGuest == nil ||
+		vmi.Status.Memory == nil ||
+		vmi.Status.Memory.GuestCurrent == nil ||
+		vmi.Status.Memory.GuestAtBoot == nil {
+		return nil
+	}
+
+	if vmi.Status.Memory.GuestCurrent.Cmp(*vmi.Spec.Domain.Memory.Guest) == 0 {
+		return nil
+	}
+	if metadata, _ := l.metadataCache.MemoryHotplug.Load(); metadata.GuestRequestedBytes == vmi.Spec.Domain.Memory.Guest.Value() {
+		return nil
+	}
+
+	requestedHotPlugMemory := vmi.Spec.Domain.Memory.Guest.DeepCopy()
+	requestedHotPlugMemory.Sub(*vmi.Status.Memory.GuestAtBoot)
+	pluggableMemoryRequested, err := vcpu.QuantityToByte(requestedHotPlugMemory)
+	if err != nil {
+		return err
+	}
+
+	spec.Devices.Memory.Target.Requested = pluggableMemoryRequested
+	memoryDevice, err := xml.Marshal(spec.Devices.Memory)
+	if err != nil {
+		log.Log.Reason(err).Error("marshalling target virtio-mem failed")
+		return err
+	}
+
+	err = dom.UpdateDeviceFlags(strings.ToLower(string(memoryDevice)), libvirt.DOMAIN_DEVICE_MODIFY_LIVE)
+	if err != nil {
+		log.Log.Reason(err).Error("updating device")
+		return err
+	}
+	log.Log.V(4).Infof("hotplugging memory for vmi %s", vmi.Name)
```

Comment:
The name is redundant as this is in the launcher.

Reply:
Will remove it.